From: Rene Kita <mail@rkta.de>
Date: Tue Sep 17 11:28:34 CEST 2024
Subject: Make geom_menu static
Patch: patches/Make-geom_menu-static.patch

While at it, remove unused function parameters.
---
 menu.c  |   10 +++++-----
 proto.h |    1 -
 2 files changed, 5 insertions(+), 6 deletions(-)

--- a/menu.c
+++ b/menu.c
@@ -343,8 +343,8 @@ new_menu(Menu *menu, MenuItem *item)
     }
 }
 
-void
-geom_menu(Menu *menu, int x, int y, int mselect)
+static void
+geom_menu(Menu *menu, int mselect)
 {
     int win_x, win_y, win_w, win_h;
 
@@ -631,7 +631,7 @@ popup_menu(Menu *parent, Menu *menu)
 	menu->cursorY = parent->cursorY;
 	guess_menu_xy(parent, menu->width, &menu->x, &menu->y);
     }
-    geom_menu(menu, menu->x, menu->y, menu->select);
+    geom_menu(menu, menu->select);
 
     CurrentMenu = menu;
     while (active) {
@@ -1495,7 +1495,7 @@ smDelBuf(char c)
     CurrentMenu->x = x;
     CurrentMenu->y = y;
 
-    geom_menu(CurrentMenu, x, y, 0);
+    geom_menu(CurrentMenu, 0);
 
     CurrentMenu->select = (mselect <= CurrentMenu->nitem - 2) ? mselect
 	: (CurrentMenu->nitem - 2);
@@ -1636,7 +1636,7 @@ smDelTab(char c)
     CurrentMenu->x = x;
     CurrentMenu->y = y;
 
-    geom_menu(CurrentMenu, x, y, 0);
+    geom_menu(CurrentMenu, 0);
 
     CurrentMenu->select = (mselect <= CurrentMenu->nitem - 2) ? mselect
 	: (CurrentMenu->nitem - 2);
--- a/proto.h
+++ b/proto.h
@@ -768,7 +768,6 @@ extern void initMouseAction(void);
 
 #ifdef USE_MENU
 extern void new_menu(Menu *menu, MenuItem *item);
-extern void geom_menu(Menu *menu, int x, int y, int mselect);
 extern void draw_all_menu(Menu *menu);
 extern void draw_menu(Menu *menu);
 extern void draw_menu_item(Menu *menu, int mselect);
